instance_selector/widgets.py

```diff
@@ -1,25 +1,16 @@
 import json
 from django.template.loader import render_to_string
 from django.urls import reverse
-from django.utils.translation import ugettext_lazy as _
-from wagtail import VERSION as WAGTAIL_VERSION
-from wagtail.admin.widgets import AdminChooser
+from django.utils.translation import gettext_lazy as _
+from wagtail.admin.widgets.chooser import AdminPageChooser
 from instance_selector.constants import OBJECT_PK_PARAM
 from instance_selector.registry import registry
 
-try:
-    from wagtail.core.telepath import register
-    from wagtail.core.widget_adapters import WidgetAdapter
-except ImportError:  # do-nothing fallback for Wagtail <2.13
+from wagtail.telepath import register
+from wagtail.widget_adapters import WidgetAdapter
 
-    def register(adapter, cls):
-        pass
 
-    class WidgetAdapter:
-        pass
-
-
-class InstanceSelectorWidget(AdminChooser):
+class InstanceSelectorWidget(AdminPageChooser):
```

Comment:
Hi Nick, upgrading M+ (one of the websites using this plugin) to Wagtail version 4.2, I used this branch to update the `wagtail-instance-selector` package.

The M+ team raised some problems wherein:

- the search function within the instance selector modal was missing
- some columns are not visible
- there is no "Add" button

Before the upgrade, the instance selector modal looked like this:
![image](https://github.com/ixc/wagtail-instance-selector/assets/26372762/93c48f04-7352-4d22-bcaa-4f6f59e50cb4)

After the upgrade, the modal looked like this:
![image](https://github.com/ixc/wagtail-instance-selector/assets/26372762/61fb805f-4813-40b6-ad70-fe02f9b7642c)

We ended up using this branch to retain the old functionalities of the modal:
https://github.com/H1ako/wagtail-instance-selector

![image](https://github.com/ixc/wagtail-instance-selector/assets/26372762/f4bb7862-9436-4023-8eae-eb22a6d59f5e)


It seems that this user forked this project after learning that this project wasn't maintained anymore (see discussion):
https://github.com/ixc/wagtail-instance-selector/pull/28#issuecomment-1328442415

I just realized that the difference is that their `InstanceSelectorWidget` [retained usage of the `AdminChooser`](https://github.com/H1ako/wagtail-instance-selector/blob/master/instance_selector/widgets.py#L18), which was [deprecated on Wagtail 4.0](https://docs.wagtail.org/en/stable/releases/4.0.html#adminchooser-replaced-with-basechooser)

With that, if we wanted to retain the old behaviour (being, able to search, add items, etc.), we might want to instead inherit from `WidgetWithScript` and `widgets.Input` much like what the deprecated `AdminChooser` did:
https://github.com/wagtail/wagtail/blob/stable/4.2.x/wagtail/admin/widgets/chooser.py#L23

What do you think?

Reply:
If we were to consider the above, since we stopped inheriting from `BaseChooser` or `AdminPageChooser`, we'll have to add the following class variables:

```
# when looping over form fields, this one should appear in visible_fields, not hidden_fields
# despite the underlying input being type="hidden"
input_type = "hidden"
is_hidden = False
```